\subsubsection*{Parameter definition}
RNA dynamics can be described by ordinary differential equations,
which have a simple analytic solution 
if the degradation and synthesis rates are assumed to be constant.
In the \verb|pulseR| package,
users need to specify the expressions for the mean RNA abundances,
Alternatively, formulas can be generated using package functions for the 
most frequent cases, see package vignettes. 

Although the most interest is focused on the gene-specific parameters,
pulseR allows to introduce shared parameters.
 This can be useful for taking into account the difference in the uridine content, since
it can introduce a bias in the estimations \citep{miller2011dynamic,
schwalb2012measurement}.
In this case, the RNA abundances are multiplied by 
a probability that at least one uridine in the molecule is substituted by 4sU.
The shared parameter then is the probability for a single base to be substituted
by a 4sU.
%\begin{equation}
%P(\text{\#u}, p) = 1-(1-p)^{\text{\#}u},
%\end{equation}
%where $p$ is the probability of a single replacement and  
% $\text{\#}u$ is the number of uridine bases in the molecule.
\subsubsection*{Normalization}
The pull-down procedure may have an effect on the amount and purity of captured RNA.
To account for difference in sequencing depth and pull-down efficiency,
we introduce additional normalization factors $\alpha_{j*}$. 
 For example, 
if the labelled fraction consists of the labelled RNA $L_{ij}$ and the unlabelled RNA
 $U_{ij}$ molecules, for a sample $j$ and gene $i$ we have 
\begin{equation}
 [\text{labelled fraction}]_{ij}  = \alpha_{j1} L_{ij} + \alpha_{j2} U_{ij}
\end{equation}
In case spike-ins are present, $\alpha_{j*}$  can be directly estimated from spike-in read counts. 
To this end, the user provides lists of spike-ins which represent unlabelled or labelled RNA.
\par
In the absence of spike-ins, 
normalization factors are derived from gene counts
because the system is overdetermined.
Inside a given RNA-seq group (e.g. [total, labelled, unlabelled ] $\times$ conditions) samples are normalized for sequencing depth $d_j$ following the DESeq procedure.
Normalization between the groups is performed during the fitting procedure, 
and these coefficients $\alpha_{G*}$  are shared between the samples from the same group $G$:
\begin{equation}
 [\text{labelled fraction}]_{ij}  = d_j(\alpha_{L1} L_{ij} + \alpha_{L2} U_{ij})
\end{equation}

\subsubsection*{Parameter estimation}
We use the maximum likelihood method (MLE) to obtain parameter values.
A typical RNA-seq experiment estimates gene abundance levels by read counts.
It has been previously shown that read counts are well represented by a negative-binomial model,
which takes over-dispersion into account 
\citep{robinson2007moderated}.
The NB distribution has two parameters, the mean $m$ and the dispersion parameter 
$r$.
Hence,  a read number of a gene $i$ in a sample $j$ follows
\begin{equation}
 K_{ij} \sim \text{NB}(m_{ij}, r).
\end{equation}
We assume $r$ to be shared between all
samples and genes. Otherwise it would not be possible to infer all parameters 
from a small number of replicates (usually, only 2 or 3 points are available).

We separated the fitting procedure into several simpler steps:
\begin{enumerate}
 \item fitting of gene-specific parameters (e.g. degradation rate)
 \item fitting of shared parameters  
 \item fitting of the normalization factors (for a spike-in-free design)
 \item estimation of the dispersion parameter 
\end{enumerate}
We repeat the steps 1-4 until user-specified convergence criteria are met.
We do not consider gene-gene interactions in this model, which 
allows us to  fit gene-specific parameters independently for every individual gene. 

We optimise the likelihood functions by using the {L-BFGS-U} method \citep{byrd1995limited}, which is
available in the \verb|stats| R package \citep{rlang}.