Gene expression abundance levels are defined by rates of RNA synthesis and degradation.
Understanding how certain gene levels are regulated by these processes across different
experimental conditions helps to gain deeper insights into RNA control mechanisms.
\par
Pulse-chase experimens facilitate to measure such kinetics \citep{wachutka2016measures}. 
Generally, "tagged" nucleoside analogs are introduced to the medium, 
taken up by cells and incorporated into nascent RNA molecules \citep{dieterich2013computational}.
For example, 4-thiouridine labelling (4sU), which was developed by \cite{dolken2008high}, is used to estimate 
kinetic rates of RNA metabolism in an increasing number of studies ( 
see \cite{wachutka2016measures} for review). Briefly, RNA labeling facilitates to separate newly synthesized from pre-existing RNA.
RNA-seq data, which are generated from sequencing these RNA pools, have a discrete nature.
To date, there is no publicly available software for kinetic parameter estimation of gene expression, which is specifically designed to handle 
fragment count data. Here we present the \verb|pulseR| package, which allows to 
process RNA-seq data from 4sU-labelling experiments.
%\citep{
%sabo2014selective,
%rabani2011metabolic,
%miller2011dynamic,
%schwanhausser2011global,
%eser2016determinants,
%schwalb2016tt,
%marzi2016degradation,
%zhang2016biogenesis,
%neymotin2014determination,
%mukherjee2016integrative}.