\subsection{Kinetic model}
\label{subsec:kinetics}
First-order reaction kinetics  is one of approaches to describe gene expression 
\citep{kaern2005stochasticity}.
Given
\begin{itemize}
 \item constant synthesis rate $s$ and
\item degradation rate $d$, 
\end{itemize}
RNA concentration $r$ follows the ordinary differential equation
\begin{equation}
 \dot{r} = s - dr,
\end{equation}
where $\dot{r}$ stands for the time derivative of the $r$.
\par
During synthesis, a new RNA molecule incorporates labelled uridine bases 
\cite{A}.
For zero initial condition $r_L(0) = 0$, the solution is
\begin{equation}
 r_\text{L}(t) = \frac{s}{d}\left(1 - e^{-dt}\right). \label{eq:labelled}
\end{equation}
With time, the labelled fraction tends to the steady state concentration
level $\mu$, 
\begin{equation}
\lim_{t\to\infty} r_\text{L}(t) = \frac{s}{d} = \mu = r_\text{T},
\end{equation}
where $r_\text{T}$ stands for the total fraction.
In contrast, the unlabelled molecules are only being degraded during the 
\emph{pulse}-experiment.
Hence, assuming initial level of unlabelled RNA to be the steady-state one, 
$r_U = \mu$,
the the amount of unlabelled fraction at a time $t$ is
\begin{equation}
 r_\text{U}(t) = \mu e^{-dt}.
\end{equation}
The example model includes only two parameters and does not consider
RNA maturation and existence of several isoforms.
For more complex approaches we refer to \cite{A}.
\par
For completeness we provide the formulas,
which describe expression levels for \emph{chase}-experiments.
In this case, we assume that no synthesis of labelled RNA occurs after 
the labelling period $t_L$:
\begin{align}
 r_\text{T}&=\mu\\
 r_\text{L}&=\mu \left(1-e^{-dt_\text{L}}\right)e^{-dt_\text{C}}\\
 r_\text{U}&=\mu \left(1-\left(1-e^{-dt_\text{L}}\right)e^{-dt_\text{C}}\right),
\end{align}
where $t_\text{C}$ stands for the longitude of the chase period.
If $t_\text{L}$ is assumed to be long enough, the ``chase''-equations above are reduced to 
\begin{align}
 r_\text{T}&=\mu\\
 r_\text{L}&=\mu e^{-dt_\text{C}}\\
 r_\text{U}&=\mu \left(1-e^{-dt_\text{C}}\right)
\end{align}
similar to the ``pulse''-equations.
\par

\subsection{Statistical model}
In RNA-seq experiments, expression level is represented by read number.
To model such data, we use the negative binomial (NB) distribution, 
because it was shown to successfully describe over-dispersed RNA-seq data 
\citep{robinson2007moderated}.
This type of distribution has two parameters, mean $m$ and dispersion parameter 
$\alpha$.
Hence,  a read number of gene $i$ in a sample $j$ reads
\begin{equation}
 K_{ij} \sim \text{NB}(m_{ij}, \alpha).
\end{equation}
Here we assume, that the dispersion parameters $\alpha$ is shared between all
samples and genes, because replicate numbers used in practice are 
very small and it's not possible to infer about several parameters 
from only 2 or 3 points.
\subsubsection{Normalisation}
\label{subsec:normalisation}		
We normalise the samples in two stages:
\begin{enumerate}
\item inside one fraction,
e.g. between samples corresponding to ``labelled, 12 hr'' measurement
\item between the fractions, 
e.g. how read numbers in the total fraction relate to the read numbers in 
``labelled, 12hr''.
\end{enumerate}
We perform the inside-fraction  normalisation as described in 
\citep{anders2010differential}. The normalisation sample-specific 
coefficient $s_j$ is the median of ratios gene read number to the geometric mean
along samples in this fraction $F$:
%\begin{equation}
% s_j = \underset{i}{\mathrm{median}}
% \frac{K_{ij}}{\left(\prod_{f \in F}K_{if}\right)^\frac{1}{|F|}}.
%\end{equation}
\begin{equation}
 s_j = \underset{i\in \text{genes}}{\mathrm{median}}
 \frac{K_{ij}}{\underset{f \in F}{\text{geometric mean }K_{if}}}.
\end{equation}
If spike-in molecules are present in the samples, then one computes $s_{j}$ 
using spike-in read numbers: 
\begin{equation}
 s_j = \underset{i \in \text{spike-ins}}{\mathrm{median}}
 \frac{K_{ij}}{\underset{f \in \text{all samples}}{\text{geometric mean }K_{if}}}.
\end{equation}
No between-fraction normalisation is needed in this case, since RNA quantities
in different fractions can be related via spike-in data only.
\par
For between-fraction normalisation, we introduce fraction-specific 
coefficients $n_f$ as model parameters, 
e.g. $n_\text{total}$ and $n_\text{label-12hr}$. The parameter values are identified
during the fitting procedure (next section).
\par
Finally, the mean $m_{ij}$ of the read number distribution is factorised as:
\begin{equation}
 m_{ij} = \text{[inside-fraction]}\times\text{[between-fractions]}\times
 r_{if} = s_jn_fr_{if},
 \end{equation}
where $r_{if}$ stands for the expression level of the gene $i$ in a fraction $f$, e.g.
$r_L$ from the eq.~\ref{eq:labelled}. 
\par
An additional relations between fractions can be introduced in the form 
of cross-contamination coefficient. For example,
if the labelled fraction $r_\text{L}$ is assumed to be 
contaminated with the unlabelled one $r_\text{U}$,
the expected read number is modified as
\begin{equation}
  r^*_\text{L} = (1-c)r_\text{L} + cr_\text{U}
 \end{equation}
\par  Such normalisation approach may look overcomplicated 
from one hand. However, the inside-fraction normalisation helps to reduce the number
of model parameters. 
\subsubsection{Fitting}
In order to estimate parameters of the model, we use the maximum-likelihood method.
We separated the fitting procedure into several simpler steps:
\begin{enumerate}
 \item fitting of gene-specific parameters (e.g. degradation rate and expression level);
 \item fitting of parameters shared between fractions (e.g. cross-contamination coefficient); 
 \item fitting of the between-sample normalisation factors $n_f$
 (if no spike-ins are provided) and estimation of the dispersion parameter $\alpha$.
\end{enumerate}
We repeat the steps 1-3 until
user-specified convergence criteria is not met.
 Since gene-gene interactions are not considered by the model, it is possible to 
 fit this parameters independently in parallel. 

To optimise the likelihood functions, 
we use implementation of the {L-BFGS-U} method \citep{byrd1995limited} 
available in the \verb|stats| R package \citep{rlang}.

\par In the paper, we visualise fitting results using \verb|ggplot2| package 
\citep{ggplot2}.